src/internal/connector/onedrive/collection.go

```diff
@@ -352,6 +359,21 @@ func (oc *Collection) getDriveItemContent(
 			return nil, clues.Wrap(err, "downloading item").Label(graph.LabelsSkippable)
 		}
 
+		// Skip big OneNote files as they can't be downloaded
+		if clues.HasLabel(err, graph.LabelStatus(http.StatusServiceUnavailable)) &&
+			oc.kind == CollectionKindPackage && *item.GetSize() >= MaxOneNoteFileSize {
```

Comment:
I'd prefer this composite check get moved as close to the original request as possible, and replaced with a clearer label like `graph.LabelsMaxFileSize`.  As a pattern, that ensures that condition identification stays centralized to the request, rather than democratized to all upstream catchers, and that the label to catch is clearly identifiable and readable.

Reply:
Firstly, about using the label `graph.LabelsMaxFileSize`. This 2GB max limit is only for OneNote files and does not affect other files(or they could have other limit) and so I don't think it is appropriate to label every file that crosses this limit with that label. Another issue is that we won't really have the size value of the item parsed at the point where we create the other labels as of now.

Another reason why this has to be moved to here instead of where the request happens is because we have to check if it is part of a `package` to determine if we should be skipping it or just failing outright.